scripts/pylib/twister/twisterlib/reports.py

```diff
@@ -296,7 +296,7 @@ def json_report(self, filename, version="NA", platform=None, filters=None):
 
         # Resolve known JSON serialization problems.
         for k,v in report_options.items():
-            report_options[k] = str(v) if type(v) in [PosixPath] else v
+            report_options[k] = str(v) if type(v) in [PosixPath, WindowsPath] else v
```

Comment:
Would be much better this way (please update imports too, and test it as I'm on my phone :))

```suggestion
            report_options[k] = str(v) if isinstance(v, Path) else v
```

Even better would be to simply use a JSON custom encoder and remove this "workaround"

```
class CustomJSONEncoder(json.JSONEncoder):
    def default(self, obj):
        if isinstance(obj, Path):
            return str(obj)

        return super().default(obj)

...

with open(filename, 'w') as json_file:
            json.dump(report, json_file, cls=CustomJSONEncoder, indent=4, separators=(',',':'))
```

Reply:
thanks, refactored.